aws-lc-rs/src/ec.rs

```diff
@@ -114,25 +114,63 @@ impl AlgorithmID {
     }
 }
 
+/// Elliptic curve public key.
 #[derive(Clone)]
-pub struct PublicKey(Box<[u8]>);
+pub struct PublicKey {
+    algorithm: &'static EcdsaSigningAlgorithm,
+    octets: Box<[u8]>,
```

Comment:
Just a general question: We were doing this previously as well, so its not a blocker. We allocate the public key with a maximum length on the stack in `marshal_public_key`, does `octets` have to be `Box` here?